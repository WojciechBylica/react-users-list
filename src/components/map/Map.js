import PropTypes from 'prop-types';
import { MapContainer, TileLayer, Marker, Popup } from 'react-leaflet';
import Error from '../error/Error';
import UserWrapper from '../user-wrapper/UserWrapper';
import '../user-wrapper/style.css';

const Map = ({ user }) => {
  const latitude = Number(user?.location?.coordinates?.latitude);
  const longitude = Number(user?.location?.coordinates?.longitude);

  if (!user?.location?.coordinates)
    return <Error message="no map data provided..." />;

  return (
    <MapContainer
      className="userWrapper userWrapper--map"
      center={[latitude, longitude]}
      zoom={3}
    >
      <UserWrapper></UserWrapper>
      <TileLayer
        attribution='&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors'
        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
      />
      <Marker position={[latitude, longitude]}>
        <Popup>
          {user?.name?.first} {user?.name?.last}
        </Popup>
      </Marker>
    </MapContainer>
  );
};
export default Map;

Map.propTypes = {
  user: PropTypes.shape({
    name: PropTypes.shape({
      first: PropTypes.string.isRequired,
      last: PropTypes.string.isRequired,
    }),
    location: PropTypes.shape({
      coordinates: PropTypes.shape({
        latitude: PropTypes.string.isRequired,
        longitude: PropTypes.string.isRequired,
      }),
    }),
  }),
};
