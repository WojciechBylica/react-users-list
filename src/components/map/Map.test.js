import React from 'react';
import renderer from 'react-test-renderer';
import Map from '.';

describe('it renders ok', () => {
  it('should render a container with map', () => {
    const tree = renderer.create(
      <Map
        user={{
          user: { name: { first: 'John', last: 'Doe' } },
          location: { coordinates: { latitude: '40', longitude: '50' } },
        }}
      />
    );
    expect(tree).toMatchSnapshot();
  });
  it('should render an error message', () => {
    const tree = renderer.create(<Map />);
    expect(tree).toMatchSnapshot();
  });
});
