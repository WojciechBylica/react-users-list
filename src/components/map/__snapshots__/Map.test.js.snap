// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders ok should render a container with map 1`] = `
<div
  className="userWrapper userWrapper--map"
/>
`;

exports[`it renders ok should render an error message 1`] = `
<p
  className="error"
  data-test="error"
>
  no map data provided...
</p>
`;
